sdcm/cluster.py

```diff
@@ -3669,6 +3672,33 @@ def verify_node_setup(start_time):
                 cl_inst.log.error(msg)
                 raise NodeSetupTimeout(msg)
 
+        @contextmanager
+        def critical_node_setup_events():
+            default_critical_events = [{'event': DatabaseLogEvent.SYSTEM_PAXOS_TIMEOUT, 'regex': r'.*Startup failed.*'}]
```

Comment:
SYSTEM_PAXOS_TIMEOUT is just one example you were seeing
but it can be from multiple types of events, depend on the failure message.

I think "Startup failed" from all of the DatabaseLogEvent should be cought here.